nupm/install.nu

```diff
@@ -23,107 +23,161 @@ def throw-error [
+            cp --recursive $mod_dir $module_dir
 
             if $package.scripts? != null {
                 log debug $"installing scripts for package ($package.name)"
-                install-scripts $path $package
+                $package.scripts | if $force {
+                    install-scripts --force $pkg_dir (script-dir --ensure)
+                } else {
+                    install-scripts $pkg_dir (script-dir --ensure)
+                }
             }
         },
         "script" => {
-            if "scripts" not-in $package {
-                let text = $"package is a script but does not have a `$.scripts` list"
-                throw-error "invalid_package_file" $text --span (metadata $path | get span)
+            log debug $"installing scripts for package ($package.name)"
+
+            $package.scripts?
+            | default [ ($pkg_dir | path join $"($package.name).nu") ]
+            | if $force {
+                install-scripts --force $pkg_dir (script-dir --ensure)
+            } else {
+                install-scripts $pkg_dir (script-dir --ensure)
             }
-
-            install-scripts $path $package
         },
         "custom" => {
-            if not ($path | path join "build.nu" | path exists) {
-                let text = $"package uses a custom install but no `build.nu` has been found"
-                throw-error "invalid_package_file" $text --span (metadata $path | get span)
+            let build_file = $pkg_dir | path join "build.nu"
+            if not ($build_file | path exists) {
+                let text = "package uses a custom install but no `build.nu` has"
+                        + " been found"
+                (throw-error
+                    "invalid_package_file"
+                    $text
+                    --span (metadata $pkg_dir | get span))
             }
 
-            nu ($path | path join 'build.nu')
+            let tmp_dir = tmp-dir build --ensure
+            cd $tmp_dir
+
+            nu $build_file ($pkg_dir | path join 'package.nuon')
+
+            cd -
+            rm -rf $tmp_dir
         },
         _ => {
-            let text = $"expected `$.type` to be one of [module, script, custom], got ($package.type)"
-            throw-error "invalid_package_file" $text --span (metadata $path | get span)
+            let text = $"expected `$.type` to be one of [module, script,"
+                + " custom], got ($package.type)"
+            (throw-error
+                "invalid_package_file"
+                $text
+                --span (metadata $pkg_dir | get span))
         },
     }
 }
+
+# Install a nupm package
+export def main [
+    name    # Name, path, or link to the package
+    --path  # Install package from a directory with package.nuon given by 'name'
+    --force(-f)  # Overwrite already installed package
+] {
+    nupm-home-prompt
+
+    if not $path {
+        throw-error "`nupm install` currently requires a `--path` flag"
+    }
```

Comment:
you do not use `--path` now, so i do not think this guard nor this option is useful now :wink:

Reply:
It's an error to indicate that `--path` is required my `nupm` to function currently. It doesn't support `nupm install some-package` at the moment.

Reply:
The `$path` is unused after the check because `--path` is currently the only install option. But as we add more, we'd add a check for `$path` and other types of installation.

Reply:
oooh ok, i think i finally got it :scream:

i was confused by the change but yeah, we installing locally by using the `name` as either the name of the remote package or a path on disk :+1: 

i think this is what confused me... :confused: 
```nushell
    name    # Name, path, or link to the package
```
maybe that would be less confusing to use
```nushell
    package    # Name, path, or link to the package
```
and then `--path` would switch from installation from the remote packages to a local one?

Reply:
@kubouch (just making sure GitHub will ping you :wink:)

Reply:
OK, I'll do that, `package` is clearer, I think. 

Yes, `--path` would force `nupm` to look for the package in your file system instead of remote.

Reply:
yeah, when i read `name`, i thought it was only the "name" of the package in some package index somewhere :open_mouth:

Reply:
Done in https://github.com/nushell/nupm/pull/18/commits/377d6ea93fc073d5a607a4d8ea4719018f33b54f

Reply:
Done in [`377d6ea` (#18)](https://github.com/nushell/nupm/pull/18/commits/377d6ea93fc073d5a607a4d8ea4719018f33b54f)